pkg/datapath/linux/node.go

```diff
@@ -1134,6 +1141,9 @@ func (n *linuxNodeHandler) NodeDelete(oldNode nodeTypes.Node) error {
 // Must be called with linuxNodeHandler.mutex held.
 func (n *linuxNodeHandler) nodeDelete(oldNode *nodeTypes.Node) error {
 	if oldNode.IsLocal() {
+		if n.nodeConfig.EnableIPSec {
+			metrics.Unregister(n.ipsecMetricCollector)
```

Comment:
Very Prometheus-naive question:
Can registering conditionally on IPsec being enabled have any lasting consequences if someone temporarily disabled IPsec? That is, can unregistering cause past data to be lost, or is the data persisted in some other way?

Reply:
AFAIK, cilium only stores metrics in memory, and didn't actually push to any external consumer. Consumers can get metric values either via cli or /metrics endpoint.

Coming back to your question, if ipsec is disabled, I think we should stop exposing any related metrics at all, as these values are no longer making sense in current setup. Additionally, this is one of the reason I configure them as Gauge type.